cdc/controller/controller.go

```diff
@@ -257,6 +258,22 @@ func (o *controllerImpl) calculateGCSafepoint(state *orchestrator.GlobalReactorS
 			forceUpdateMap[upstreamID] = nil
 		}
 	}
+
+	// check if the upstream has a changefeed, if not we should update the gc safepoint
+	_ = o.upstreamManager.Visit(func(up *upstream.Upstream) error {
+		hasGcServiceSafePoint := false
+		if _, exist := minCheckpointTsMap[up.ID]; exist {
+			hasGcServiceSafePoint = true
+		}
+		if _, exist := forceUpdateMap[up.ID]; exist {
```

Comment:
This is a redundant judgment since all upstreamID related information is stored in minCheckpointTsMap.